src/NuGet.Core/NuGet.Configuration/Properties/AssemblyInfo.cs

```diff
@@ -15,3 +15,4 @@
 [assembly: InternalsVisibleTo("NuGet.Commands.FuncTest, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
 [assembly: InternalsVisibleTo("Test.Utility, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
 [assembly: InternalsVisibleTo("NuGet.Commands.Test, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
+[assembly: InternalsVisibleTo("Microsoft.Build.NuGetSdkResolver, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
```

Comment:
Why is this needed?

If the NuGet SDK resolver needs APIs from NuGet.Configuration, then as much as I hate excessive public APIs, it seems to be that it should be a public API. Using `InternalsVisibleTo` for reasons other than testing feels like an anti-pattern, and makes it far more likely that the target assembly will be able to accidentally take on dependencies on APIs in Configuration that really should be internal only.

Reply:
Because of `SettingsLoadContext.FileRead`. IMO, if your proposal for declaring a provider in NuGet.Common is accepted, then this can go away too, since SettingsLoadContext.FileRead can report an event itself directly.

Reply:
Yeah alternatively I could make `SettingsLoadingContext.FileRead` public, I don't think it would really have any ramifications as its just an event that's fired and not something an end user would directly call.  We just could never change the signature...